Docs: Update Grafana Teams documentation (#86942)
grafana/grafana: docs/sources/administration/team-management/configure-grafana-teams.md

Reviewer: In general, its preferable to use administrator instead of admin.

---

> aliases:
>   - 

Reviewer: ```suggestion
```

description: Learn how to configure Grafana Teams
labels:
  products:
    - enterprise
    - oss
    - cloud
keywords:
  - authorization
  - microservices
  - architecture

> menuTitle: Configure  Grafana Teams

Reviewer: ```suggestion
menuTitle: Configure Grafana Teams
```

title: Configure Grafana Teams
weight: 200
---

# Configure Grafana Teams

This topic describes how to set up and configure Grafana Teams.

> For a tutorial on working with Teams see [Create users and teams](https://grafana.com/tutorials/create-users-and-teams/).

Reviewer: https://grafana.com/docs/writers-toolkit/write/style-guide/style-conventions/#links-and-references

```suggestion
For a tutorial on working with Teams, refer to [Create users and teams](https://grafana.com/tutorials/create-users-and-teams/).
```

## Before you begin

Before you begin creating and working with Grafana Teams:

> - Ensure that you have either the `Organization Administrator` role or team administrator permissions. Refer to [Organization roles](https://grafana.com/docs/grafana/latest/administration/roles-and-permissions/#organization-roles) and [RBAC permissions, actions, and scopes](https://grafana.com/docs/grafana/latest/administration/roles-and-permissions/access-control/custom-role-actions-scopes/#rbac-permissions-actions-and-scopes) for a list of Grafana roles and role-based access control actions.
> - Decide which users belong to which teams and what permissions team members receive.
> - Configure the default basic role for users to join Grafana. This role applies to users where no role is set by the identity provider (IDP).
>   - No basic role - by default cannot view any resources. Recommended for `isolated` teams.
>   - Viewer role - by default can view all resources. Recommended for `collaborative` teams.
> - Ensure team sync is turned on if you plan to manage team members through team sync. Refer to [Configure Team Sync](https://grafana.com/docs/grafana/latest/setup-grafana/configure-security/configure-team-sync/)  for  a list of providers and instructions on how to turn on team sync for each provider.

Reviewer: ```suggestion
- Ensure that you have either the `Organization Administrator` role or team administrator permissions.
  Refer to [Organization roles](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/administration/roles-and-permissions/#organization-roles) and [RBAC permissions, actions, and scopes](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/administration/roles-and-permissions/access-control/custom-role-actions-scopes/#rbac-permissions-actions-and-scopes) for a list of Grafana roles and role-based access control actions.
- Decide which users belong to which teams and what permissions team members receive.
- Configure the default basic role for users to join Grafana. This role applies to users where no role is set by the identity provider (IDP).
  - No basic role - by default cannot view any resources.
      Recommended for `isolated` teams.
  - Viewer role - by default can view all resources. Recommended for `collaborative` teams.
- Ensure team sync is turned on if you plan to manage team members through team sync.
  Refer to [Configure Team Sync](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/setup-grafana/configure-security/configure-team-sync/)  for  a list of providers and instructions on how to turn on team sync for each provider.
```

<!-- - Turn on nested folders.  __This is a new feature, add when it goes live.__ -->

{{< admonition type="note" >}}

> [Grafana Organizations](https://grafana.com/docs/grafana/latest/administration/organization-management/) do not exist in Grafana Cloud. Grafana Cloud uses the term “organization” to refer to accounts in grafana.com, or GCOM. In Grafana Enterprise and OSS, Teams belong to Grafana Organizations. Refer to [About organizations](/docs/grafana/latest/administration/organization-management/#about-organizations) for more information.

Reviewer: ```suggestion
[Grafana Organizations](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/administration/organization-management/) do not exist in Grafana Cloud.
Grafana Cloud uses the term _organization_ to refer to accounts in grafana.com, or GCOM.
In Grafana Enterprise and OSS, Teams belong to Grafana Organizations.
Refer to [About organizations](/docs/grafana/<GRAFANA_VERSION>/administration/organization-management/#about-organizations) for more information.
```

To add a role, complete the following steps:

1. Sign in to Grafana as an `org administrator` or `team administrator`.
1. Click the arrow next to **Administration** in the left-side menu, click **Users and access**, and select **Teams**. 
1. Select the Team and click under the **Role** column. Select from a list of current fixed or plugin roles or clear all roles and start over. As you hover over each role a list of permissions appears to the right. You can uncheck any permission for additional fine-grained control.
1. Click **Update** to add the new role or roles. 

To delete a role, remove the check next to the role name and click **Update**. 

## Delete a team

Delete a team when you no longer need it. This action permanently deletes the team and removes all team permissions from dashboards and folders. This task requires that you have `organization administrator` permissions.

1. Sign in to Grafana as an `org administrator` or `team administrator`.
1. Click the arrow next to **Administration** in the left-side menu, click **Users and access**, and select **Teams**. 
1. Click the **red X** on the right side of the name of the team.
1. Click **Delete**.

> ## Create isolated or collaborative teams

Reviewer 2: I would remove the Create isolated or collaborative teams section for now.
Going to work with @IevaVasiljeva to prepare an example in terraform that should be easier to follow than this code snipet.

Author: @sarahleejane I can comment this out for now.

Author: done

Grafana Teams can either be `isolated` or `collaborative`. Isolated teams can only see their own resources. They cannot see other team’s dashboards, data, alerts, etc. Collaborative teams have access to other team’s resources. Grafana Cloud users must contact Support.

To create an isolated team add the following to the Grafana configuration file:

> ```

Reviewer: For syntax highlighting, the Grafana configuration file is INI.

````suggestion
```ini
````

auto_assign_org_role = None
role_attribute_path = contains(groups[*], 'admin') && 'Admin' || 'None'
```

To create a collaborative team add the following to the Grafana configuration file:

> ```

Reviewer: ````suggestion
```ini
````

auto_assign_org_role = Viewer
role_attribute_path = contains(groups[*], 'admin') && 'Admin' || 'None'
```
You can also use a terraform script as shown in the following example:

> ```terraform

Reviewer: For syntax highlighting, you actual want `hcl` which stands for Hashicorp Configuration Language.

````suggestion
```hcl
````

keywords:
  - authorization
  - microservices
  - architecture
menuTitle: Configure Grafana Teams
title: Configure Grafana Teams
weight: 200
---

# Configure Grafana Teams

This topic describes how to set up and configure Grafana Teams.

For a tutorial on working with Teams, refer to [Create users and teams](https://grafana.com/tutorials/create-users-and-teams/).

## Before you begin

Before you begin creating and working with Grafana Teams:

- Ensure that you have either the `Organization Administrator` role or team administrator permissions.
  Refer to [Organization roles](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/administration/roles-and-permissions/#organization-roles) and [RBAC permissions, actions, and scopes](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/administration/roles-and-permissions/access-control/custom-role-actions-scopes/#rbac-permissions-actions-and-scopes) for a list of Grafana roles and role-based access control actions.
- Decide which users belong to which teams and what permissions team members receive.
- Configure the default basic role for users to join Grafana. This role applies to users where no role is set by the identity provider (IDP).

>   - No basic role - by default cannot view any resources.

Reviewer: nit: contractions

```suggestion
  - No basic role - by default can't view any resources.
```

1. Click the name of the Team to which you want to add members, and click **+ Add member**.
1. Search for and select a user.
1. Choose whether to add the user as a Team **Member** or **Admin**.
1. Click **Save**.

## Grant or change Team member permissions

Complete this task when you want to add or modify team member permissions.

To grant team member permissions:

1. Sign in to Grafana as an `org administrator` or `team administrator`.
1. Click the arrow next to **Administration** in the left-side menu, click **Users and access**, and select **Teams**.
1. Click the pencil next to the name of the Team for which you want to add or modify team member permissions.
1. In the team member list, locate the user that you want to change. You can use the search field to filter the list if necessary.
1. Under the **Permission** column, select the new permission level.

## Add roles to a Grafana Team

You can add or delete roles from a specified team.

To add a role, complete the following steps:

1. Sign in to Grafana as an `org administrator` or `team administrator`.
1. Click the arrow next to **Administration** in the left-side menu, click **Users and access**, and select **Teams**.

> 1. Select the Team and click under the **Role** column. Select from a list of current fixed or plugin roles or clear all roles and start over. As you hover over each role a list of permissions appears to the right. You can uncheck any permission for additional fine-grained control.

Reviewer: https://developers.google.com/style/word-list#uncheck

```suggestion
1. Select the Team and click under the **Role** column.
   Select from a list of current fixed or plugin roles or clear all roles and start over.
   As you hover over each role a list of permissions appears to the right.
   You can clear any permission for additional fine-grained control.
```
